python/torch_mlir/eager_mode/torch_mlir_dispatch.py

```diff
@@ -0,0 +1,406 @@
+            # e.g. this happens for SliceOutOfLowerBoundEndIndexModule
+            # i.e. when you slice down to a zero dim (by going out of bounds)
+            # NB this is actually extra bad because mlir::makeStridedLinearLayoutMap
+            # fails an assert
+            shape = map(lambda x: x or - 1, arg.shape)
+            annotations.append(
+                # we need torch dtype here rather than np dtype in order to pass to ModuleBuilder
+                TorchTensorType(shape=tuple(shape), dtype=normalized_args[i].dtype)
+            )
+            tensor_args.append(arg)
+
+    # pull out tensor kwargs and put them in positional order
+    tensor_kwargs_flat = []
+    if unwrapped_normalized_kwargs:
+        tensor_kwargs = {}
+        arg_idxs = {
+            arg_name: i
+            for i, arg_name in enumerate(
+                [arg.name for arg in script_fun.schema.arguments]
+            )
+        }
+        for i, (kw, arg) in enumerate(unwrapped_normalized_kwargs.items()):
+            if isinstance(arg, np.ndarray):
+                # we need torch dtype here rather than np dtype in order to pass to ModuleBuilder
+                tensor_kwargs[arg_idxs[kw]] = (arg, normalized_kwargs[kw].dtype)
+
+        for i in range(len(tensor_kwargs)):
+            arg, arg_dtype = tensor_kwargs[i]
+            annotations.append(TorchTensorType(shape=tuple(arg.shape), dtype=arg_dtype))
+            tensor_kwargs_flat.append(arg)
+
+    return annotations, tensor_args, tensor_kwargs_flat
+
+
+op_registry = load_registry()
+
+
+def try_torch_mlir_eager(op, args, kwargs, backend):
+    if hasattr(op, "op_name"):
+        op_name = op.op_name
+    elif hasattr(op, "__name__"):
+        # builtin_function_or_method
+        op_name = op.__name__
+
+    if op_name == "detach":
+        raise RuntimeError("detaching")
+    if "layout" in kwargs and kwargs["layout"] is not None:
+        # TODO: e.g. ones_like comes in with a sparse strided layout
+        warnings.warn(f"{kwargs['layout']} layout not supported; changing to default layout.")
+        kwargs["layout"] = None
+    if "memory_format" in kwargs and kwargs["memory_format"] is not None:
+        warnings.warn(f"{kwargs['layout']} memory format not supported; changing to default memory format.")
+        kwargs["memory_format"] = None
+
+    schema, normalized_args, normalized_kwargs = normalize_op(op, args, kwargs)
+    script_fun = build_script_function(schema, normalized_args, normalized_kwargs)
+    annotations, tensor_args, tensor_kwargs_flat = annotate_args_kwargs(
+        script_fun, normalized_args, normalized_kwargs
+    )
+    func_annotation = Annotation(annotations)
+    mb = ModuleBuilder()
+    mb.import_function(script_fun)
+    eager_module = build_module(mb, script_fun, func_annotation)
+
+    qualified_op_name = f"{schema.name}" + (
+        f".{schema.overload_name}" if schema.overload_name else ""
+    )
+    # TODO: replace with no repro report passman
```

Comment:
Can you address this TODO? If this is still useful for debugging adding a global "enable_debug_mode" could be useful.